spdx/checksum.go

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+package spdx
+
+type ChecksumAlgorithm2_2 int
```

Comment:
If a checksumAlgorithm type is set as an int, we loose the ability to distinguish the algorithm type in runtime unless we compare the algorithm type against all the possible algorithms.

The point I'm trying to put across is that, we would be identifying each ChecksumAlgorithm2_2 as an int. If we were to extent this model to let the ChecksumAlgorithm2_2 object decide it's name, it won't be inherently possible. This also would break the implicit typecasting.

For example: 
```go
fmt.Println(Checksum2_2{
	Algorithm: SHA1,
	Value:     "",
})
```

This would print `{0 }`

To resolve this, there can either be two approaches:
1. Add another function of ChecksumAlgorithm2_2 that maps the algorithm type to their string equivalent.
2. Let the enum type be a string instead of int.